static/src/javascripts/projects/common/modules/commercial/dfp/apply-creative-template.js

```diff
@@ -0,0 +1,110 @@
+    'common/modules/commercial/creatives/fluid250',
+    'common/modules/commercial/creatives/fluid250GoogleAndroid',
+    'common/modules/commercial/creatives/hosted-thrasher',
+    'common/modules/commercial/creatives/hosted-thrasher-multi',
+    'common/modules/commercial/creatives/scrollable-mpu',
+    'common/modules/commercial/creatives/scrollable-mpu-v2',
+    'common/modules/commercial/creatives/template'
+], function (
+    bean,
+    qwery,
+    Promise,
+    fastdom
+) {
+    /**
+     * Not all adverts render themselves - some just provide data for templates that we implement in commercial.js.
+     * This looks for any such data and, if we find it, renders the appropriate component.
+     */
+    return function applyCreativeTemplate($adSlot) {
+        return getAdvertIframe($adSlot).then(function (iframe) {
+            return renderCreativeTemplate($adSlot, iframe);
+        });
+    };
+
+    function getAdvertIframe($adSlot) {
+        return new Promise(function (resolve, reject) {
+            var iframes = qwery('iframe', $adSlot);
+            var nonEmptyFrames = iframes.filter(function (iframe) {
+                // DFP will sometimes return empty iframes, denoted with a '__hidden__' parameter embedded in its ID.
+                // These never reach 'complete' state in IE, so we filter them.
+                return iframe.id.match('__hidden__') === null;
+            });
+            var contentFrame = nonEmptyFrames[0];
+
+            if (!contentFrame) {
+                reject();
+            }
+            // On IE, wait for the frame to load before interacting with it
+            else if (contentFrame.readyState && contentFrame.readyState !== 'complete') {
+                bean.on(contentFrame, 'readystatechange', function (e) {
+                    var updatedIFrame = e.srcElement;
+
+                    if (
+                        /*eslint-disable valid-typeof*/
+                    updatedIFrame &&
+                    typeof updatedIFrame.readyState !== 'unknown' &&
+                    updatedIFrame.readyState === 'complete'
+                    /*eslint-enable valid-typeof*/
+                    ) {
+                        bean.off(updatedIFrame, 'readystatechange');
+                        resolve(contentFrame);
+                    }
+                });
+            } else {
+                resolve(contentFrame);
+            }
+        });
+    }
+
+    function renderCreativeTemplate($adSlot, iFrame) {
+        var creativeConfig = fetchCreativeConfig();
+
+        if (creativeConfig) {
+            return Promise.all([
+                renderCreative(creativeConfig),
+                hideIframe()
+            ]);
+        } else {
+            return Promise.resolve();
+        }
+
+        function fetchCreativeConfig() {
+            var breakoutScript = iFrame.contentDocument.body.querySelector('.breakout__script[type="application/json"]');
```

Comment:
Apparently that was just a prototype; our surveys actually just use custom creatives in DFP, instead. These use images rather than text, so they don't need to be broken out to use branded Guardian fonts.